File: project-{topic}-advanced/backend/src/models/task.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');
const User = require('./user'); // Import User for association
const Project = require('./project'); // Import Project for association

const Task = sequelize.define('Task', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
    allowNull: false
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
    validate: {
      notEmpty: true,
      len: [3, 150]
    }
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  status: {
    type: DataTypes.ENUM('to-do', 'in-progress', 'done'),
    defaultValue: 'to-do',
    allowNull: false
  },
  priority: {
    type: DataTypes.ENUM('low', 'medium', 'high'),
    defaultValue: 'medium',
    allowNull: false
  },
  dueDate: {
    type: DataTypes.DATE,
    allowNull: true
  },
  projectId: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: Project,
      key: 'id'
    }
  },
  assignedTo: {
    type: DataTypes.UUID,
    allowNull: true, // Can be null if not assigned to a specific user yet
    references: {
      model: User,
      key: 'id'
    }
  },
  createdAt: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  },
  updatedAt: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'tasks',
});

// Associations
Task.belongsTo(Project, { foreignKey: 'projectId', as: 'project' });
Task.belongsTo(User, { foreignKey: 'assignedTo', as: 'assignee' });
Task.belongsTo(User, { foreignKey: 'creatorId', as: 'creator' }); // Assuming a creatorId will be added or implicitly through auth

module.exports = Task;
```

### `backend/src/models/index.js` (Model Associations and Export)
```javascript

File: project-{topic}-optimized/app/schemas/task.py
from typing import Optional
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime, date

class TaskBase(BaseModel):
    title: str = Field(min_length=3, max_length=150)
    description: Optional[str] = Field(None, max_length=1000)
    status: str = Field("pending", pattern="^(pending|in-progress|completed)$")
    due_date: Optional[date] = None

class TaskCreate(TaskBase):
    project_id: UUID
    assignee_id: Optional[UUID] = None

class TaskUpdate(TaskBase):
    title: Optional[str] = Field(None, min_length=3, max_length=150)
    status: Optional[str] = Field(None, pattern="^(pending|in-progress|completed)$")
    assignee_id: Optional[UUID] = None
    project_id: Optional[UUID] = None # Allow changing project for a task

class TaskInDBBase(TaskBase):
    id: UUID
    project_id: UUID
    assignee_id: Optional[UUID] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Task(TaskInDBBase):
    pass
```